### PR TITLE
xplat: match posix / unix date output and fix JIT Date issue.

### LINKED ISSUE #1593: xplat: Date/formatting.js fails with jit on
With "-Test" build, there is one weird PDT/PST error (from mktime call) during the loop of test cases. Excluded in #1591.

### PR DESCRIPTION
**Pull request by obastemur**
Fixes #1593 

Initially xplat posix Date implementation was matching (at least it was trying!) with Windows OldDateAPI. However, we are unable to meet some of the edge cases + number inconsistencies among the distros.

Remove all the bad hacks on posix imp. and match results to posix / unix default instead + add an additional test case.

### CODE REVIEW

File: test/Date/rlexe.xml [hunk 1]
```diff
@@ -87,14 +87,20 @@
...
+  </test>
+  <test>
+    <default>
       <files>formatting.js</files>
       <baseline>formatting.baseline</baseline>
       <compile-flags>-ForceOldDateAPI</compile-flags>
       <!-- test is timezone-sensitive; remove exclude_jenkins after fix -->
       <!-- on DST pass Win OldDateAPI jumps back to 01:00 after 01:59 -->
       <!-- todo: Do not force OLDDateAPI ? -->
       <!-- xplat: on Linux Test build (post JIT) mktime returns PST within series of PDT -->
```
Comment (obastemur) [inline test/Date/rlexe.xml]: self note: remove this comment

**Comment by obastemur** [comment]
@dotnet-bot test Ubuntu ubuntu_linux_debug_static please

File: lib/Runtime/PlatformAgnostic/Platform/Linux/DateTime.cpp [hunk 2]
```diff
@@ -193,23 +190,16 @@ namespace DateTime
...
-        // right in between dst pass
-        // we need mktime trick to get the correct dst
-        utc_tm.tm_isdst = 1;
-        ltime = mktime(&utc_tm);
-        ltime += utc_tm.tm_gmtoff;
-        localtime_r(&ltime, &utc_tm);
-
-        isDaylightSavings = utc_tm.tm_isdst;
-        offset = utc_tm.tm_gmtoff / 60;
+        isDaylightSavings = local_tm.tm_isdst ? 0 : 1;
```
Comment (jianchun) [inline lib/Runtime/PlatformAgnostic/Platform/Linux/DateTime.cpp]: >isDaylightSavings = local_tm.tm_isdst ? 0 : 1; [](start = 8, length = 46)

"isDaylightSavings" is of type "bool", why assign "0/1" instead of "false/true"?

**Comment by jianchun** [comment]
:shipit:

**Comment by obastemur** [comment]
@jianchun Thanks for the review
